eCourse_backend/courses/views.py

```diff
@@ -26,26 +26,52 @@ def course_student_list_iframe(request, name, page=1):
                 first_name__contains=filter_form['first_name'].data,
                 last_name__contains=filter_form['last_name'].data,
                 username__contains=filter_form['username'].data)
-    else:
-        filter_form = CourseStudentFilterForm()
-        students = Student.objects.all()
-
-    paginator = Paginator(students, 10)
-    page_obj = paginator.get_page(page)
+    elif request.method == 'GET':
+        filter_form = CourseStudentFilterForm() 
+        course = get_object_or_404(Course, pk = id)
+        lecturer = course.lecturer
+        # print('lecturer', lecturer)
+        students = course.student.all()
 
     context = {
-        'page_obj': page_obj,
         'filter_form': filter_form,
+        'lecturer': lecturer,
+        'students': students,
     }
     return render(request, 'courses/iframes/course_student_list.html', context)
 
 
+@login_required
+def course_overview(request, page=1):
+    user_id = request.user.id
+    if request.method == 'GET':
+        # officer has type 3 in db
+        if request.user.type == 3:
+            courses = Course.objects.all()
+        else:
+            # courses = Course.objects.filter(student = user_id)
+            courses = Course.objects.filter(Q(student = user_id) | Q(lecturer_id = user_id))
+    
+    paginator = Paginator(courses, 10)
+    page_obj = paginator.get_page(page)
+
+    context = {
+        'page_obj': page_obj,
+    }   
+    return render(request, 'courses/overview.html', context)
+
 @login_required
 @permission_required('courses.create_course', raise_exception=True)
```

Comment:
i don't think we need the permission create_course for the overview function. As it is written right now only lecturers and officers can view this. But this is something students also should see

Reply:
This View is from the backend and isn't used by the frontend, so it shouldn't matter for this PR

Reply:
yes you are right but i think is is something we can fix by removing one line of code and that is in my oppinion something that can be done befor merging to the master branch